mesonbuild/modules/pkgconfig.py

```diff
@@ -78,33 +80,39 @@ class GenerateKw(TypedDict):
     listify=True,
 )
 
+
+def _as_str(obj: object) -> str:
+    assert isinstance(obj, str)
+    return obj
+
+
 class DependenciesHelper:
     def __init__(self, state: ModuleState, name: str) -> None:
         self.state = state
         self.name = name
-        self.pub_libs = []
-        self.pub_reqs = []
-        self.priv_libs = []
-        self.priv_reqs = []
-        self.cflags = []
-        self.version_reqs = {}
-        self.link_whole_targets = []
-
-    def add_pub_libs(self, libs) -> None:
-        libs, reqs, cflags = self._process_libs(libs, True)
-        self.pub_libs = libs + self.pub_libs # prepend to preserve dependencies
+        self.pub_libs: T.List[LIBS] = []
+        self.pub_reqs: T.List[str] = []
+        self.priv_libs: T.List[LIBS] = []
+        self.priv_reqs: T.List[str] = []
+        self.cflags: T.List[str] = []
+        self.version_reqs: T.DefaultDict[str, T.Set[str]] = defaultdict(set)
+        self.link_whole_targets: T.List[T.Union[build.CustomTarget, build.CustomTargetIndex, build.StaticLibrary]] = []
+
+    def add_pub_libs(self, libs: T.Sequence[ANY_DEP]) -> None:
+        p_libs, reqs, cflags = self._process_libs(libs, True)
```

Comment:
... right here we add
```python
assert isinstance(libs, list), 'for mypy'
````

Which we should do anyway, because process_libs should not conceptually take `self._process_libs('foo', True)`, that will break.